### PR TITLE
Add vprintf support

### PR DESCRIPTION
**Pull request by rydeom**
Its nothing special, but its nice so have (I needed it lol)

### CODE REVIEW

File: core/include/printf.h
Comment (geissdoerfer) [inline core/include/printf.h:1]: Please don't change the formatting of this file. It's an external library and I would like to be able to diff it with a new version.

File: core/runtime.c
Comment (geissdoerfer) [inline core/runtime.c:1]: Are these just the changes that are already on the no-cap-mon branch? In that case, I would just merge them from there.

File: linker.ld [hunk 1]
```diff
@@ -147,6 +147,7 @@ SECTIONS {
                 *(.teardown)
                 __teardown_end__ = .;
                 __bss_end__ = .;
+                end = __bss_end__;
```
Comment (geissdoerfer) [inline linker.ld]: Where is this label needed? Why is there no _start_ label, but just an end label?

**Review by geissdoerfer — CHANGES_REQUESTED** [review]
Please remove the newer two commits from this PR -  they are not related to the vprintf support. I will merge the no-cap-mon branch now, so this functionality becomes available. Finally, please create a separate PR for the missing label and explain in the comments, why this is needed. Thanks!

**Comment by rydeom** [comment]
Sorry for the messed up stuff. Now its only the important changes.

**Review by geissdoerfer — APPROVED** [review]

**Pull request merged**